### PR TITLE
Add PreUpgrade check - Disable custom DNS

### PR DESCRIPTION
**Pull request by mjudeikis**
### Which issue this PR addresses:

Fixes cases when customer enables custom DNS but never rotated nodes, and we rotate those during the upgrade and :boom: 

### What this PR does / why we need it:

This fails to upgrade if custom DNS is set. 

### Test plan for issue:

Create a cluster in custom VNET
Enable custom DNS
Restart one of the nodes to see if they get broken
Execute ` curl -k -X PATCH      -H 'Content-Type: application/json'      "https://localhost:8443/subscriptions/${AZURE_SUBSCRIPTION_ID}/resourcegroups/${RESOURCEGROUP}/providers/Microsoft.RedHatOpenShift/openShiftClusters/${CLUSTER}?api-version=admin" -d {}`
Profit 

### Is there any documentation that needs to be updated for this PR?

N/A

### CODE REVIEW

**Comment by mjudeikis** [comment]
We should convert this to "check before upgrade" so we do not rotate the nodes if this is set. I the future we would propagate reason to SupportabilityProfile.

File: pkg/util/azureclient/mgmt/network/virtualnetworks_addons.go [hunk 1]
```diff
@@ -11,9 +11,19 @@ import (
 
 // VirtualNetworksClientAddons contains addons for VirtualNetworksClient
 type VirtualNetworksClientAddons interface {
+	CreateOrUpdateAndWait(ctx context.Context, resourceGroupName string, virtualNetworkName string, parameters mgmtnetwork.VirtualNetwork) (err error)
 	List(ctx context.Context, resourceGroupName string) (virtualnetworks []mgmtnetwork.VirtualNetwork, err error)
 }
```
[the remark this answers is not in the document]
Comment (mjudeikis) [reply pkg/util/azureclient/mgmt/network/virtualnetworks_addons.go]: Good catch. Thanks

**Comment by ms-jasondel** [comment]
Is ARO planning to support setting custom DNS on the ARO VNet? Currently this does not work with an error:
"custom DNS servers are not supported" when attempting to deploy a cluster.

**Pull request merged**